lib/utils/jsontools.go

```diff
@@ -114,8 +114,26 @@ func FastMarshalIndent(v interface{}, prefix, indent string) ([]byte, error) {
 	return data, nil
 }
 
-// WriteJSON marshals multiple documents as a JSON list with indentation.
-func WriteJSON(w io.Writer, values interface{}) error {
+// WriteJSONArray marshals values as a JSON array.
+func WriteJSONArray[T any](w io.Writer, values []T) error {
+	if len(values) == 0 {
+		_, err := w.Write([]byte("[]"))
+		return err
+	}
+	return writeJSON(w, values)
+}
+
+// WriteJSONObject marshals m as a JSON object.
+func WriteJSONObject[M ~map[K]V, K comparable, V any](w io.Writer, m M) error {
+	if len(m) == 0 {
+		_, err := w.Write([]byte("[]"))
```

Comment:
An object should be written as `{}` not `[]`

Reply:
Updated to `{}`